src/components/settings/SignerAccountMFA/useMFASettings.ts

```diff
@@ -0,0 +1,32 @@
+import useMPC from '@/hooks/wallets/mpc/useMPC'
+import { COREKIT_STATUS } from '@web3auth/mpc-core-kit'
+
+export type MFASettings = {
+  mfaEnabled: boolean
+} | null
+
+const useMFASettings = () => {
+  const mpcCoreKit = useMPC()
+
+  if (mpcCoreKit?.status !== COREKIT_STATUS.LOGGED_IN) {
+    return null
+  }
+
+  const { shareDescriptions } = mpcCoreKit?.getKeyDetails()
+
+  const hashedShareModuleFactor = Object.entries(shareDescriptions).find(([key, value]) =>
+    value[0]?.includes('hashedShare'),
+  )
```

Comment:
Nit: you already return early in `SignerAccountMFA` if the status is `!COREKIT_STATUS.LOGGED_IN`. I'd suggest moving this logic to a helper function and calling it after the early return in `SignerAccountMFA` instead.